### PR TITLE
Support top/bottom/left/right drawer direction

### PR DESCRIPTION
**Pull request by dcyoung**
I needed support for a drawer that pulls in from the right for use w/ landscape tablets. This effectively extends an existing open PR https://github.com/emilkowalski/vaul/pull/183

Haven't thoroughly tested the snap-points or CSS logic, but it worked for my needs.

Includes a modified example in the test app, w/ a select for opening each direction.

### CODE REVIEW

**Comment by IgnisDa** [comment]
@emilkowalski Any updates on this?

**Comment by theboxer** [comment]
Would be amazing to have this one in.

File: test/src/app/with-scaled-background/page.tsx [hunk 1]
```diff
@@ -1,100 +1,94 @@
 'use client';
 
+import classNames from 'classnames';
```
Comment (emilkowalski) [inline test/src/app/with-scaled-background/page.tsx:3]: Can we use clsx instead of classnames? It's already installed.

File: src/context.ts [hunk 2]
```diff
@@ -45,11 +47,12 @@ export const DrawerContext = React.createContext<DrawerContextValue>({
   modal: false,
   shouldFade: false,
   activeSnapPoint: null,
-  onOpenChange: () => {},
-  setActiveSnapPoint: () => {},
+  onOpenChange: () => { },
```
Comment (emilkowalski) [inline src/context.ts:50]: Prettier

**Comment by emilkowalski** [comment]
Lockfile is frozen, should be fine after you revert the addition of `classnames`

**Comment by gruckion** [comment]
I need left and right 🤣

**Pull request merged**